code/modules/reagents/reagent_dispenser.dm

```diff
@@ -390,4 +390,106 @@
+
+/obj/structure/reagent_dispensers/cauldron/barrel/update_icon()
+	return
+
+/obj/structure/reagent_dispensers/cauldron/barrel/kick_act(mob/living/carbon/human/H)
+	..()
+	if (!reagents)
+		return 1
+	if(reagents.total_volume > 10) //Beakersplashing only likes to do this sound when over 10 units
+		playsound(src, 'sound/effects/slosh.ogg', 25, 1)
+	usr.investigation_log(I_CHEMS, "has emptied \a [src] ([type]) containing [reagents.get_reagent_ids(1)] onto \the [usr.loc].")
+	reagents.reaction(usr.loc)
+	src.reagents.clear_reagents()
+	usr.visible_message("<span class='warning'>[usr] kicks \the [src]!</span>", "<span class='notice'>You kick \the [src].</span>")
+	for(var/atom/movable/AM in src)
+		AM.forceMove(loc)
+
+/obj/structure/reagent_dispensers/cauldron/barrel/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	if(iswrench(W))
+		return
+	if(istype(W,/obj/item/weapon/grab))
+		var/obj/item/weapon/grab/G = W
+		var/mob/living/target = G.affecting
+		user.visible_message("<span class='danger'>[user] begins to drag [target] into the barrel!</span>")
+		if(do_after_many(user,list(target,src),10)) //Twice the normal time
+			enter_barrel(target)
+	else
+		..()
+	return ..()
+
+/obj/structure/reagent_dispensers/cauldron/barrel/attackby(obj/item/W, mob/user, params)
+
+/obj/structure/reagent_dispensers/cauldron/barrel/proc/enter_barrel(mob/user)
+	user.forceMove(src)
+	update_icon()
+	user.reset_view()
+	to_chat(user,"<span class='notice'>You enter \the [src].</span>")
+
+/obj/structure/reagent_dispensers/cauldron/barrel/MouseDropTo(atom/movable/O, mob/user)
+	if(O.loc == user || !isturf(O.loc) || !isturf(user.loc) || !user.Adjacent(O)) //no you can't pull things out of your ass
+		return
+	if(user.incapacitated() || user.lying) //are you cuffed, dying, lying, stunned or other
+		return
+	if(!Adjacent(user) || !user.Adjacent(src)) // is the mob too far away from you, or are you too far away from the source
+		return
+	if(O.locked_to)
+		return
+	else if(O.anchored)
+		return
+	if(issilicon(O)) //robutts dont fit
+		return
+	if(!ishigherbeing(user) && !isrobot(user)) //No ghosts or mice putting people into the barrel
+		return
+	var/mob/living/target = O
+	if(!istype(target))
+		return
+	for(var/mob/living/carbon/slime/M in range(1,target))
+		if(M.Victim == target)
+			to_chat(usr, "[target.name] will not fit into the [src] because they have a slime latched onto their head.")
+			return
+
+	if(target == user)
+		to_chat(user,"<span class='notice'>You begin to climb into the barrel.</span>")
+		if(do_after(target,src,10))
+			enter_barrel(target)
+	else
+		user.visible_message("<span class='danger'>[user] begins to drag [target] into the barrel!</span>")
+		if(do_after_many(user,list(target,src),10)) //Twice the normal time
+			enter_barrel(target)
+
+/obj/structure/reagent_dispensers/cauldron/barrel/container_resist(mob/user)
+	if(exiting.Find(user))
```

Comment:
Instead of `Find()`ing then `-=`'ing, you can use `exiting.Remove(user)` to search and remove, it returns 1 if it succeeded